### PR TITLE
adjust jenkins template setting to account for effects of constrained…

### PR DESCRIPTION
**Pull request by gabemontero**
… default max heap

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1551500
Fixes https://github.com/openshift/jenkins/issues/501

@openshift/sig-developer-experience ptal

### CODE REVIEW

File: examples/jenkins/jenkins-ephemeral-template.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
       "metadata": {
         "name": "${JENKINS_SERVICE_NAME}",
         "annotations": {
-          "template.openshift.io/expose-uri": "http://{.spec.host}{.spec.path}"
+          "template.openshift.io/expose-uri": "http://{.spec.host}{.spec.path}",
+	  "haproxy.router.openshift.io/timeout": "2m"
```
Comment (bparees) [inline examples/jenkins/jenkins-ephemeral-template.json]: alignment

File: examples/jenkins/jenkins-ephemeral-template.json [hunk 2]
```diff
@@ -90,15 +91,16 @@
...
                   "initialDelaySeconds": 3,
                   "httpGet": {
                     "path": "/login",
                     "port": 8080
                   }
                 },
                 "livenessProbe": {
-                    "timeoutSeconds": 3,
+                    "timeoutSeconds": 240,
+		    "periodSeconds": 360,
```
Comment (bparees) [inline examples/jenkins/jenkins-ephemeral-template.json]: alignment

File: examples/jenkins/jenkins-persistent-template.json [hunk 3]
```diff
@@ -26,7 +26,8 @@
       "metadata": {
         "name": "${JENKINS_SERVICE_NAME}",
         "annotations": {
-          "template.openshift.io/expose-uri": "http://{.spec.host}{.spec.path}"
+          "template.openshift.io/expose-uri": "http://{.spec.host}{.spec.path}",
+	  "haproxy.router.openshift.io/timeout": "2m"
```
Comment (bparees) [inline examples/jenkins/jenkins-persistent-template.json]: alignment

File: examples/jenkins/jenkins-persistent-template.json [hunk 4]
```diff
@@ -107,15 +108,16 @@
...
                   "initialDelaySeconds": 3,
                   "httpGet": {
                     "path": "/login",
                     "port": 8080
                   }
                 },
                 "livenessProbe": {
-                    "timeoutSeconds": 3,
+                    "timeoutSeconds": 240,
+		    "periodSeconds": 360,
```
Comment (bparees) [inline examples/jenkins/jenkins-persistent-template.json]: alignment

**Comment by bparees** [comment]
@sdodson fyi, we're going to need to pull this change into openshift-ansible once it merges.

@gabemontero please make this change in here too: https://github.com/openshift/jenkins/tree/master/openshift/templates  (once my work finishes, you'll only need to make the change in the jenkins repo and it will filter down from jenkins->library->origin, but we're in an in between state right now).

**Comment by gabemontero** [comment]
10-4 @bparees

**Comment by gabemontero** [comment]
There is also a lot of history and diagnostic details in https://github.com/openshift/jenkins/pull/440

**Comment by gabemontero** [comment]
also @sdodson this is the master branch PR

given the current state of master, I created https://github.com/openshift/origin/pull/18834 for the 3.9 branch

File: examples/jenkins/jenkins-ephemeral-template.json [hunk 5]
```diff
@@ -90,15 +91,16 @@
...
                     "path": "/login",
                     "port": 8080
                   }
                 },
                 "livenessProbe": {
-                    "timeoutSeconds": 3,
+                    "timeoutSeconds": 240,
+		    "periodSeconds": 360,  
                     "initialDelaySeconds": 420,
                     "failureThreshold" : 30,
```
Comment (bparees) [inline examples/jenkins/jenkins-ephemeral-template.json]: with a polling interval of 6 mins, i think the failure threshold (number of failures required before we consider it dead) needs to be reduced.

**Comment by gabemontero** [comment]
updates pushed @bparees

**Comment by gabemontero** [comment]
fyi closed out 3.9 https://github.com/openshift/origin/pull/18834

will create a new 3.9 PR off of this commit once this one gets approved

**Pull request merged**